account_mtd/controllers/main.py

```diff
@@ -0,0 +1,40 @@
+import base64
+import logging
+from odoo import http, exceptions
+from datetime import datetime, timedelta
+from werkzeug.utils import redirect
+
+
+class Authorize(http.Controller):
+    _logger = logging.getLogger(__name__)
+    
+    @http.route('/auth-redirect', type='http', methods=['GET'])
+    def get_user_authorization(self, **args):
+        
+        # to determine which API the authorization code has been received for.
+        api_tracker = http.request.env['mtd.api_request_tracker'].search([('closed', '=', False)])
+        if len(api_tracker) != 1:
```

Comment:
Should this not read more like:
```
if not api_tracker:
    logger.warn("...")
    raise exceptions.Warning()
elif len(api_tracker) > 1:
    logger.info("...")
    ....
    do something
    ...
else:
    for record in api_tracker:
    ....
    do something
    ....
```
so that handling for none, one, and more than one is seperate? I'm just suggesting this based on the comment below about `user should never be in a state where they found no tracker record`